app/components/navpatientheader/navPatientHeaderHelpers.js

```diff
@@ -0,0 +1,18 @@
+import personUtils from '../../core/personutils';
+import _ from 'lodash';
+
+export const getPermissions = (patient, permsOfLoggedInUser) => {
+  const permissions = patient.permissions;
+
+  const canUpload = _.isEmpty(permissions) === false && permissions.root || _.has(permsOfLoggedInUser, 'upload');
+  const canShare = _.isEmpty(permissions) === false && permissions.root;
+
+  return { canUpload, canShare };
+}
+
+export const getPatientListLink = (clinicFlowActive, selectedClinicId, query) => {
+  let patientListLink = clinicFlowActive && selectedClinicId ? '/clinic-workspace/patients' : '/patients';
+  if (query?.dashboard) patientListLink = `/dashboard/${query.dashboard}`;
+
+  return { patientListLink }
```

Comment:
These conditions copied from `app/components/navbar/navbar.js:37` with some changes:

-- `showPatientListLink` is irrelevant since we now always want to show a back button
-- `isDashboard` is not irrelevant because we do not show this on the navbar.